app/src/main/java/org/simple/clinic/ViewControllerBinding.kt

```diff
@@ -92,6 +92,7 @@ class ViewControllerBinding<E, S, T>(
 
     disposable = uiEvents
         .compose(uiStateProducer)
+        .share()
```

Comment:
@vinaysshenoy you may have to look into this. This fixes a problem where the upstream pipeline has multiple subscribers. In our case, a `doOnNext` call was invoked once for every subscription.